Add issues experienced during test upgrade (#1822)
stackhpc/stackhpc-kayobe-config: doc/source/operations/upgrading-openstack.rst

Known issues
============

Pulp Upgrade
------------

It was found that if pulp is upgraded from 3.43.1 to 3.81.0, most of pulp
APIs become unusable because of a bug related to missing response header
when docker clients expect it.

Generally pulp is considered to be safe to completely destroy and re-deploy.
So, users are encouraged to upgrade pulp in this method.

If your pulp has custom built images, we recommended either not doing pulp

> upgrade by pinning pulp version at ``seed_pulp_container.pulp.tag``in

Reviewer: ```suggestion
upgrade by pinning pulp version at ``seed_pulp_container.pulp.tag`` in
```

Author: oof

Known issues
============

Pulp Upgrade
------------

It was found that if pulp is upgraded from 3.43.1 to 3.81.0, most of pulp
APIs become unusable because of a bug related to missing response header
when docker clients expect it.

Generally pulp is considered to be safe to completely destroy and re-deploy.
So, users are encouraged to upgrade pulp in this method.

If your pulp has custom built images, we recommended either not doing pulp
upgrade by pinning pulp version at ``seed_pulp_container.pulp.tag``in
``$KAYOBE_CONFIG_PATH/seed.yml`` (``$KAYOBE_CONFIG_PATH/environments/<env>/seed.yml``
if using environments) to 3.43.1,

or pulling all custom built images before destroying pulp then push them again
after pulp upgrade is done.

Let's Encrypt
-------------

`Let's Encrypt TLS settings fix <https://review.opendev.org/c/openstack/kolla-ansible/+/925971>`__
brought breaking change to Let’s Encrypt ansible role. Now users have to explictly set the target
Let’s Encrypt ACME server as a kolla ansible variable ``letsencrypt_external_cert_server``/
``letsencrypt_internal_cert_server`` if they were using Let’s Encrypt as a CA of their

> external/internal TLS certificates.

Reviewer: ```suggestion
`Let's Encrypt TLS settings fix <https://review.opendev.org/c/openstack/kolla-ansible/+/925971>`__
brought a breaking change to Let’s Encrypt ansible role. Now users have to explicitly set the target
Let’s Encrypt ACME server as a kolla ansible variable ``letsencrypt_external_cert_server``/
``letsencrypt_internal_cert_server`` if they were using Let’s Encrypt as a CA of their
external/internal TLS certificates.
```

Cinder
------

`Enhancement of Ceph integration of multiple clusters
<https://review.opendev.org/c/openstack/kolla-ansible/+/907166>`__
made Cinder role now requires ``user`` and ``pool`` set to the each item of kolla dict
variable ``cinder_ceph_backends`` at ``$KAYOBE_CONFIG_PATH/kolla/globals.yml``
(``$KAYOBE_CONFIG_PATH/environments/<env>/kolla/globals.yml`` if using environments)

> For example,

Reviewer: ```suggestion
`Enhancement of Ceph integration of multiple clusters
<https://review.opendev.org/c/openstack/kolla-ansible/+/907166>`__
means the Cinder role now requires ``user`` and ``pool`` set to the each item of kolla dict
variable ``cinder_ceph_backends`` at ``$KAYOBE_CONFIG_PATH/kolla/globals.yml``
(``$KAYOBE_CONFIG_PATH/environments/<env>/kolla/globals.yml`` if using environments)
For example,
```

.. code:: yaml

   cinder_ceph_backends:
      - name: rbd-1
         cluster: ceph
         user: cinder
         pool: volumes
         enabled: true
      - name: rbd-2
         cluster: ceph-hdd
         user: cinder
         pool: volumes-hdd
         enabled: true

> You can find the name of pools from ``cephadm_pools`` in cephadm.yml.

Reviewer: What about `user`?

Author: It's gonna be ``cinder`` unless configured to be different before. I assume no one did as this could not be customised.

> The K-A upstream change `<https://review.opendev.org/c/openstack/kolla-ansible/+/909974>`__

Reviewer: Would be best to shorten this
```suggestion
The K-A upstream change `#909974 <https://review.opendev.org/c/openstack/kolla-ansible/+/909974>`__
```

requires users to manually set Cinder cluster name at ``$KAYOBE_CONFIG_PATH/kolla/globals.yml``
(``$KAYOBE_CONFIG_PATH/environments/<env>/kolla/globals.yml`` if using environments).

> It should be matching with current cluster name. You can find the name from ``cinder.conf``.

Reviewer: I'm not 100% sure that this _is_ the correct variable, but we should definitely include the variable name here.
```suggestion
It should match the current cluster name. You can find the name from ``cinder.conf`` as ``rbd_cluster_name``.
```

Author: Good point. It's actually ``cluster`` in [DEFAULT]. I'll include that.

.. code:: yaml

   cinder_cluster_name: ceph

CloudKitty
----------

Elasticseach storage driver is no longer compatible with Opensearch storage backend.
Need to set CloudKitty storage backend to Opensearch if it was set to be Elasticseach before.
This can be set at ``$KAYOBE_CONFIG_PATH/kolla/globals.yml``

> (``$KAYOBE_CONFIG_PATH/environments/<env>/kolla/globals.yml`` if using environments)

Reviewer: ```suggestion
The elasticsearch storage driver is no longer compatible with Opensearch storage backend.
Set the CloudKitty storage backend to ``opensearch`` if it was ``elasticseach`` before.
This can be set at ``$KAYOBE_CONFIG_PATH/kolla/globals.yml``
(``$KAYOBE_CONFIG_PATH/environments/<env>/kolla/globals.yml`` if using environments)
```

.. code:: yaml

   cloudkitty_storage_backend: opensearch

Ironic
------

From Dalmatian, `Kayobe no longer provides its own default driver & interfaces
<https://review.opendev.org/c/openstack/kayobe/+/836999>`__
for Ironic and follows Ironic's default.

> This can cause your Ironic configuration ``ironic.conf`` to have regression.

Reviewer: ```suggestion
This can cause your Ironic configuration ``ironic.conf`` to regress.
```
